platforms/minigame/platforms/taobao-mini-game/wrapper/engine/cache-manager.js

```diff
@@ -0,0 +1,110 @@
+ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ THE SOFTWARE.
+ ****************************************************************************/
+ const { deleteFile, rmdirSync, unzip, isOutOfStorage } = require('../fs-utils');
+ (function(){
+    var unzipSuffix = 0;
+    if (!(cc && cc.assetManager && cc.assetManager.cacheManager)) {
+        return;
+    }
+    
+    const cacheManager = cc.assetManager.cacheManager;
+    var cleaning = false;
+
+    cacheManager.clearLRU = function() {
+        if(cc.downloadFileNumForTest){
+            cc.downloadFileNumForTest = 0;
+        }
+        if (cleaning) return;
+        cleaning = true;
+        var caches = [];
+        var self = this;
+        if(this.tempFiles && this.tempFiles.length > 0){
+            this.tempFiles.forEach(function (val, key) {
+                var time = Date.now().toString();
+                caches.push({ originUrl: key, url: val, lastTime: time });
+            });
+        }
+        if(this.tempFiles && this.tempFiles.length > 0){
+            this.cachedFiles.forEach(function (val, key) {
+                if (val.bundle === 'internal') return;
+                if (self._isZipFile(key) && cc.assetManager.bundles.find(bundle => bundle.base.indexOf(val.url) !== -1)) return;
+                caches.push({ originUrl: key, url: val.url, lastTime: val.lastTime });
+            });
+        }
+        caches.sort(function (a, b) {
+            return a.lastTime - b.lastTime;
+        });
+        caches.length = Math.floor(caches.length / 3);
+        if (caches.length === 0) {
+            cleaning = false;
+            console.warn('can not get the file system!');
+            return;
+        }
+        for (var i = 0, l = caches.length; i < l; i++) {
+            this.cachedFiles.remove(caches[i].originUrl);
+        }
+        
+        this.writeCacheFile(function () {
+            function deferredDelete () {
+                var item = caches.pop();
+                if (self._isZipFile(item.originUrl)) {
+                    deleteFile(item.url, self._deleteFileCB.bind(self));
+                }
+                else {
+                    deleteFile(item.url, self._deleteFileCB.bind(self));
+                }
+                if (caches.length > 0) { 
+                    setTimeout(deferredDelete, self.deleteInterval); 
+                }
+                else {
+                    cleaning = false;
+                }
+            }
+            setTimeout(deferredDelete, self.deleteInterval);
+        });
+
+    };
+    cacheManager.unzipAndCacheBundle = function (id, zipFilePath, cacheBundleRoot, onComplete) {
+        let time = Date.now().toString();
+        let targetPath = `${this.cacheDir}/${cacheBundleRoot}/${time}${unzipSuffix++}`;
+        let self = this;
+        unzip(zipFilePath, targetPath, function (err) {
```

Comment:
TargetPath maybe unexist. Does it need to create the target directory firstly?

Reply:
in taobao platform ,unzip function will create the directory automatically.